src/arch/mod.rs

```diff
@@ -83,6 +89,7 @@ impl StructType {
 }
 
 impl Type {
+    /// Returns true if `other` can be cast to `self` without losing infomation.
```

Comment:
```suggestion
    /// Returns true if `other` can be converted to `self` without losing infomation.
```

Cast makes me think an explicit cast (`(int)1.4`), I think conversion is better.